File: sdk/php/src/Resources/CephS3.php
<?php

namespace HiTechCloud\SDK\Resources;

class CephS3 extends BaseResource
{
    /**
     * Get connection info
     */
    public function getConnectionInfo(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/s3');
    }

    /**
     * Get credentials
     */
    public function getCredentials(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/s3/credentials');
    }

    /**
     * Get usage
     */
    public function getUsage(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/s3/usage');
    }

    /**
     * Get metrics
     */
    public function getMetrics(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/s3/metrics');
    }

    /**
     * List buckets
     */
    public function listBuckets(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/s3/buckets');
    }

    /**
     * Create bucket
     */
    public function createBucket(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/s3/buckets', $data);
    }

    /**
     * Delete bucket
     */
    public function deleteBucket(int $serviceId, string $bucket): array
    {
        return $this->http->delete('/api/service/' . $serviceId . '/s3/buckets/' . $bucket);
    }

    /**
     * List sub-users
     */
    public function listSubusers(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/s3/subusers');
    }

    /**
     * Create sub-user
     */
    public function createSubuser(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/s3/subusers', $data);
    }

    /**
     * Delete sub-user
     */
    public function deleteSubuser(int $serviceId, string $subuser): array
    {
        return $this->http->delete('/api/service/' . $serviceId . '/s3/subusers/' . $subuser);
    }

    /**
     * Rotate secret key
     */
    public function rotateSecretKey(int $serviceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/s3/key');
    }

}

File: sdk/csharp/Resources/CephS3.cs
namespace HiTechCloud.SDK.Resources;

public class CephS3 : BaseResource
{
    public CephS3(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> S3Async(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/s3", id));
    }

    public async Task<Dictionary<string, object?>> CredentialsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/s3/credentials", id));
    }

    public async Task<Dictionary<string, object?>> UsageAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/s3/usage", id));
    }

    public async Task<Dictionary<string, object?>> MetricsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/s3/metrics", id));
    }

    public async Task<Dictionary<string, object?>> BucketsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/s3/buckets", id));
    }

    public async Task<Dictionary<string, object?>> CreateBucketAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/s3/buckets", id), data);
    }

    public async Task<Dictionary<string, object?>> DeleteBucketAsync(int id, string bucket)
    {
        return await Http.DeleteAsync(string.Format("/api/service/{id}/s3/buckets/{bucket}", id, bucket));
    }

    public async Task<Dictionary<string, object?>> SubusersAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/s3/subusers", id));
    }

    public async Task<Dictionary<string, object?>> CreateSubuserAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/s3/subusers", id), data);
    }

    public async Task<Dictionary<string, object?>> DeleteSubuserAsync(int id, string subuser)
    {
        return await Http.DeleteAsync(string.Format("/api/service/{id}/s3/subusers/{subuser}", id, subuser));
    }

    public async Task<Dictionary<string, object?>> RotateKeyAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/s3/key", id), data);
    }
}
